blockchain/ethereum.go

```diff
@@ -170,6 +170,29 @@ func (e *EthereumClient) GetWallets() []*EthereumWallet {
 	return e.Wallets
 }
 
+// NewWallet generates a new ethereum wallet and adds it to the Wallets list, funding it if funding is specified
+// and returning its index in the wallet list
+func (e *EthereumClient) NewWallet(funding *big.Int) (int, error) {
+	privateKey, err := crypto.GenerateKey()
+	if err != nil {
+		return -1, err
+	}
+	privateKeyBytes := crypto.FromECDSA(privateKey)
+	privateKeyHex := hex.EncodeToString(privateKeyBytes)
+	newWallet, err := NewEthereumWallet(privateKeyHex)
+	if err != nil {
+		return -1, err
+	}
+	if funding != nil {
+		err = e.Fund(newWallet.Address(), conversions.WeiToEther(funding), GasEstimations{})
+		if err != nil {
+			return -1, err
+		}
```

Comment:
As there is no data, can do inline error check.
```suggestion
		if err = e.Fund(newWallet.Address(), conversions.WeiToEther(funding), GasEstimations{}); err != nil {
			return -1, err
		}
```

Reply:
Applied, Ty